### PR TITLE
fix: potential use of mismatched tokens

### PR DESCRIPTION
**Pull request by pepesi**
related issue https://github.com/alibaba/higress/issues/1082

### CODE REVIEW

**Comment by CH3CHO** [comment]
Is this ready for review now? @pepesi

**Comment by pepesi** [comment]
> Is this ready for review now? @pepesi

sorry，I haven't fully tested it yet

**Comment by CH3CHO** [comment]
> > Is this ready for review now? @pepesi
> 
> sorry，I haven't fully tested it yet

No problem. Please take your time.

File: plugins/wasm-go/extensions/ai-proxy/provider/moonshot.go [hunk 1]
```diff
@@ -55,10 +55,12 @@ func (m *moonshotProvider) OnRequestHeaders(ctx wrapper.HttpContext, apiName Api
 	if apiName != ApiNameChatCompletion {
 		return types.ActionContinue, errUnsupportedApiName
 	}
+	apiKey := m.config.GetRandomToken()
```
Comment (CH3CHO) [inline plugins/wasm-go/extensions/ai-proxy/provider/moonshot.go:58]: 感觉可以在加一个基于ctx获取token的方法。如果ctx有，那就直接返回。如果没有，就随机一个放到ctx里。

Comment (CH3CHO) [reply plugins/wasm-go/extensions/ai-proxy/provider/moonshot.go:58]: 这样这个方法谁都可以用。

Comment (pepesi) [reply plugins/wasm-go/extensions/ai-proxy/provider/moonshot.go]: 添加了   config.GetOrSetTokenWithContext 方法，暂时还没时间测试，我会在晚些时候找到另外一个账号的apikey测试 后再回复，预计周五前。

**Review by CH3CHO — DISMISSED** [review]
LGTM. Thanks.

**Comment by pepesi** [comment]
Sorry, I am busy with other things and will handle it as soon as possible

**Comment by CH3CHO** [comment]
@pepesi Please fix the conflicts. Thanks.

**Review by CH3CHO — APPROVED** [review]
LGTM. Thanks.

**Pull request merged**